www/src/components/ColorModePicker.tsx

```diff
@@ -0,0 +1,68 @@
+import { useState } from 'react';
+
+const storageKey = 'recharts-color-mode';
+
+type ColorModeStrategy = 'light' | 'dark' | 'system';
+
+function getStoredColorModeStrategy(): ColorModeStrategy {
+  const rawStoredValue = localStorage.getItem(storageKey);
+  if (rawStoredValue == null) {
+    return 'system';
+  }
+  if (rawStoredValue !== 'dark' && rawStoredValue !== 'light') {
+    return 'system';
+  }
+  return rawStoredValue;
+}
+function setStoredColorModeStrategy(strategy: ColorModeStrategy) {
+  localStorage.setItem(storageKey, strategy);
+}
+function clearStoredColorModeStrategy() {
+  localStorage.removeItem(storageKey);
+}
+
+function setColorMode(strategy: ColorModeStrategy) {
+  if (strategy === 'system') {
+    const prefersDark = window.matchMedia('(prefers-color-scheme: dark)').matches;
+    document.documentElement.setAttribute('data-mode', prefersDark ? 'dark' : 'light');
+  } else {
+    document.documentElement.setAttribute('data-mode', strategy);
+  }
+}
+
+function setInitialColorMode() {
+  const strategy = getStoredColorModeStrategy();
+  setColorMode(strategy);
+}
+
+export const ColorModePicker = Object.assign(
```

Comment:
why are we using Object.assign? Just export each function/component and use it as is?